module/zfs/gzip.c

```diff
@@ -56,6 +57,15 @@ gzip_compress(void *s_start, void *d_start, size_t s_len, size_t d_len, int n)
 
 	ASSERT(d_len <= s_len);
 
+#ifdef HAVE_QAT
+	if (qat_init_done) {
```

Comment:
Because we can only take advantage of QAT when the source buffer is large, ```qat_init_done``` can be revised to a function and it does:
```
bool use_qat(size_t s_len)
{
    if (!qat_init_done || s_len <= PAGE_SIZE)
        return (false);
    return (true);
}
```
I just write ```PAGE_SIZE``` as an example, this value should be figured out by testing - if the buffer is less than this size, the benefit of using QAT will be wiped out by the overhead of communicating with QAT devices.